### ISSUE TITLE
[Feature Request] Add peertube support

### ISSUE THREAD

**Issue by stemy2** [ajayyy/SponsorBlock#1209]
Hello,

Sometimes i watch videos on peertube, and since SB is not compatible with it, i'm not able to skip sponsored sequences.

**Comment by erkinalp** [comment]
As PeerTube is federated, SponsorBlock would need to keep track of which instance is which. The current query method is not that suited to that.

**Comment by farOverNinethousand** [comment]
While I myself do not need/want PeerTube support, the problem of keeping track which instance a video is being played on could be solved by adding the domain of that instance into the videoID e.g. "peertubedomain.tld:videoID".
...though again if once instance has multiple domains this could lead to issues and for sure one wouldn't want to keep track of peertube instances' domains/main domains manually.

...though generally that PeerTube request makes sense as PeerTube support would automatically support a wide range of websites using peertube though I can imagine some issues could arise with:
- different/older revisions of peertube
- heavily modified peertube instances

**Comment by ajayyy** [comment]
Domain is not needed for peertube, the IDs are unique, and can appear on multiple instances due to federation

**Comment by erkinalp** [comment]
> Domain is not needed for peertube, the IDs are unique, 

It is, if you also want to de-federated instances.

**Comment by farOverNinethousand** [comment]
IDs look like some 

> But the id is still globally unique

That would definitely be the easiest case.

**Comment by erkinalp** [comment]
@ajayyy There is no such guarantee across any two instances that do not federate with each other.

**Comment by raphj** [comment]
I ran into this issue this week. Some channels are both on PeerTube and YouTube. I'd rather use and encourage PeerTube, but I find sponsored parts unbearable and I'm often listening to videos with no immediate access to playback controls.

Metadata gathered from one source could be used for the other source by letting users declare that a given PeerTube video is the same as some given YouTube video or vice-versa (which is most certainly easily automatable, though this automation is easily outside the scope of SponsorBlock). SponsorBlock would need to store a mapping or sets of equivalent video ids (this could also be useful for duplicate YouTube video I guess, but I'm not sure this is very useful).

Would that be conceivable?

**Comment by Uneccessary** [comment]
> As PeerTube is federated, SponsorBlock would need to keep track of which instance is which. The current query method is not that suited to that.

Sponsorblock already keeps track of Invidious and Piped instances, which is basically the same thing.

Also, there are public listings of PeerTube instances which can be used.

**Comment by jblemee** [comment]
For peertube video imported from youtube, there is addon: https://github.com/jblemee/peertube-plugin-sponsorblock

**Comment by stemy2** [comment]
Interesting, but it must be implemented by instance admins themselves, am i right?

**Comment by jblemee** [comment]
Yes it must be installed by instance admin and it works only with imported youtube video. so, not a full workarround
